packages/dashboard-message-bus-common/lib/messages.ts

```diff
@@ -87,6 +105,7 @@ export const isInvalidateMessage = (
 };
 
 export type MessageType =
+  | CliEventMessage
```

Comment:
:( goodbye generic param

Reply:
Oh should I specify a param here? It defaults to `unknown` so I omitted it.

Reply:
It's unfortunate... you'll note that if you try to specify a param, you'll have to add it to the main `MessageType` type, where it doesn't really belong (since it's not like `T` applies elsewhere)

But basically, I hate generic defaults because you can very quickly lose track of the parameter without realizing it (like here... anything that uses `MessageType` has no way of specifying what kind of CliEvent payload)

Reply:
This comment is not to say we should change this... @benjamincburns makes a good argument for the use of generics in the first place, it's just you end up with stuff like this unless you do a whole bunch of work up front.

(This sort of thing is why I was saying not to bother and instead just keep making new types, but I'm happy to lose that argument if this example is the worst of it.)

Reply:
Ah I see. Would this be better?

```ts
// Rid default
interface CliEventMessage<T> extends Message { // ...

// Specify param
type MessageType = CliEventMessage<unknown> | // ...
```

Reply:
Up to you if you want to keep the default... if I were you, I'd go either way, but I probably would take care not to rely on the default personally (like do at least "specify param" if not "rid default")

Reply:
I see, I removed the default param and let `MessageType` specify `unknown`. Update [here](https://github.com/trufflesuite/truffle/pull/5621/commits/822a619edf169dc12874d7fe9ba262dbe3754f47).

I think this isn't very important at the moment: `MessageType` isn't actually used anywhere in the dashboard package, only its extended interfaces are.